shipyard/src/main/java/org/jclouds/shipyard/domain/servicekeys/ServiceKey.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jclouds.shipyard.domain.servicekeys;
+
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.json.SerializedNames;
+
+import com.google.auto.value.AutoValue;
+
+@AutoValue
+public abstract class ServiceKey {
+
+   public abstract String key();
+   
+   @Nullable public abstract String description();
+      
+   ServiceKey() {
+   }
+
+   @SerializedNames({ "key", "description" })
+   public static ServiceKey create(String key, String description) {
+      return new AutoValue_ServiceKey(key, description);
+   }
```

Comment:
I'd declare the Create and Delete service key classes as inner classes here. They don't seem to make sense outside the scope of this class, so I'd configure them as inner classes.